controller/controller.go

```diff
@@ -99,39 +100,111 @@ func (c *Controller) Attach(attachRequest map[string]string) resources.FlexVolum
+		Message:    "Volume is attached",
+		VolumeName: volumeName,
+		Device:     volumeName,
+		Attached:   true,
+	}
+}
+
+//WaitForAttach Waits for a volume to get attached to the node
+func (c *Controller) WaitForAttach(waitForAttachRequest k8sresources.FlexVolumeWaitForAttachRequest) k8sresources.FlexVolumeResponse {
+	c.logger.Println("controller-waitForAttach-start")
+	defer c.logger.Println("controller-waitForAttach-end")
+	return k8sresources.FlexVolumeResponse{
+		Status:     "Success",
+		Message:    "Volume attached",
+		VolumeName: waitForAttachRequest.Name,
+		Device:     waitForAttachRequest.Name,
+		Attached:   true,
+	}
+}
+
+//IsAttached checks if volume is attached
+func (c *Controller) IsAttached(isAttachRequest k8sresources.FlexVolumeIsAttachedRequest) k8sresources.FlexVolumeResponse {
+	c.logger.Println("controller-isAttached-start")
+	defer c.logger.Println("controller-isAttached-end")
+	return k8sresources.FlexVolumeResponse{
+		Status:     "Success",
+		Message:    "Volume is attached",
+		VolumeName: isAttachRequest.Name,
+		Device:     isAttachRequest.Name,
+		Attached:   true,
+	}
+}
+
 //Detach detaches the volume/ fileset from the pod
-func (c *Controller) Detach(detachRequest resources.FlexVolumeDetachRequest) resources.FlexVolumeResponse {
+func (c *Controller) Detach(detachRequest k8sresources.FlexVolumeDetachRequest) k8sresources.FlexVolumeResponse {
 	c.logger.Println("controller-detach-start")
 	defer c.logger.Println("controller-detach-end")
 
 	c.logger.Printf("detach-details %#v. (no operation done in the detach action)\n", detachRequest)
 
-	// TODO : no-op for now, will change with latest flex api update
+	// TODO : no-op for now, will change with latest k8sresources api update
 
-	return resources.FlexVolumeResponse{
+	return k8sresources.FlexVolumeResponse{
 		Status:  "Success",
 		Message: "Volume detached successfully",
 		Device:  detachRequest.Name,
 	}
 }
 
+//MountDevice mounts a device in a given location
+func (c *Controller) MountDevice(mountDeviceRequest k8sresources.FlexVolumeMountDeviceRequest) k8sresources.FlexVolumeResponse {
+	c.logger.Println("controller-MountDevice-start")
+	defer c.logger.Println("controller-MountDevice-end")
+	c.logger.Printf("mountDeviceRequest %#v", mountDeviceRequest)
+
+	mountRequest := k8sresources.FlexVolumeMountRequest{MountDevice: mountDeviceRequest.Name, MountPath: mountDeviceRequest.Path, Opts: mountDeviceRequest.Opts}
+	//return k8sresources.FlexVolumeResponse{
+	//	Status:  "Success",
+	//	Message: "Volume is mounted",
+	//}
+	return c.Mount(mountRequest)
+}
+
+//UnmountDevice checks if volume is unmounted
+func (c *Controller) UnmountDevice(unmountDeviceRequest k8sresources.FlexVolumeUnmountDeviceRequest) k8sresources.FlexVolumeResponse {
+	c.logger.Println("controller-UnmountDevice-start")
+	defer c.logger.Println("controller-UnmountDevice-end")
+	c.logger.Printf("mountDeviceRequest %#v", unmountDeviceRequest)
+	return k8sresources.FlexVolumeResponse{
```

Comment:
now its confusing
in the mountdevice API you executed the mount function, so i expect that here in the umountdevice API you will run umount function.
could you please explain why no umount is triggered here?